modules/siddhi-extensions/event-table/src/test/java/org/wso2/siddhi/extension/eventtable/hazelcast/InsertIntoTableTestCase.java

```diff
@@ -0,0 +1,458 @@
+                "@from(eventtable = 'hazelcast', instance.name = 'siddhi_instance')" +
+                "define table StockTable (symbol string, price float, volume long); " +
+                "@from(eventtable = 'hazelcast', instance.name = 'siddhi_instance')" +
+                "define table StockTable2 (symbol string, price float, volume long); ";
+        String query = "" +
+                "@info(name = 'query1') " +
+                "from StockStream " +
+                "insert into StockTable ;" +
+                "" +
+                "@info(name = 'query2') " +
+                "from StockStream2 " +
+                "insert into StockTable2 ;";
+
+        ExecutionPlanRuntime executionPlanRuntime = siddhiManager.createExecutionPlanRuntime(streams + query);
+        InputHandler stockStream = executionPlanRuntime.getInputHandler("StockStream");
+
+        executionPlanRuntime.start();
+        stockStream.send(new Object[]{"WSO2", 55.6f, 100l});
+        stockStream.send(new Object[]{"IBM", 75.6f, 100l});
+        stockStream.send(new Object[]{"WSO2", 57.6f, 100l});
+        Thread.sleep(RESULT_WAIT);
+        executionPlanRuntime.shutdown();
+    }
+
+    @Test
+    public void insertIntoTableTest4() throws InterruptedException {
+        log.info("InsertIntoTableTest4");
+
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String streams = "" +
+                "define stream StockStream (symbol string, price float, volume long); " +
+                "define stream StockCheckStream (symbol string); " +
+                "@from(eventtable = 'hazelcast', instance.name = 'siddhi_instance')" +
+                "define table StockTable (symbol string, price float, volume long); ";
+        String query = "" +
+                "@info(name = 'query1') " +
+                "from StockStream " +
+                "insert into StockTable ;" +
+                "" +
+                "@info(name = 'query2') " +
+                "from StockCheckStream[symbol==StockTable.symbol in StockTable] " +
+                "insert into OutStream ;";
+
+        ExecutionPlanRuntime executionPlanRuntime = siddhiManager.createExecutionPlanRuntime(streams + query);
+
+        executionPlanRuntime.addCallback("query2", new QueryCallback() {
+            @Override
+            public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
+                EventPrinter.print(timeStamp, inEvents, removeEvents);
+                if (inEvents != null) {
+                    for (Event event : inEvents) {
+                        inEventCount.incrementAndGet();
+                        switch (inEventCount.get()) {
+                            case 1:
+                                Assert.assertArrayEquals(new Object[]{"WSO2"}, event.getData());
+                                break;
+                            default:
+                                Assert.assertSame(1, inEventCount.get());
```

Comment:
Can you refactor these assertions as well so that the assertion failure happens within the test case thread?